src/http/headers.rs

```diff
@@ -0,0 +1,137 @@
+use crate::ffi::{
+    ngx_http_headers_in_t, ngx_http_headers_out_t, ngx_list_part_t, ngx_list_t, ngx_pool_t, ngx_table_elt_t,
+    ngx_uint_t, u_char,
+};
+use crate::ngx_string;
+use std::ffi::CString;
+use std::fmt;
+use std::ptr;
+
+pub struct Header(*mut ngx_table_elt_t, *mut ngx_pool_t);
+
+impl Header {
+    pub fn set_value(&mut self, value: &str) {
+        // we can use pool to allocate and then copy
+        unsafe { self.0.as_mut() }.map(|table| {
+            table.value.len = value.len() as _;
+            table.value.data = crate::ffi::str_to_uchar(self.1, value);
+        });
+
+        // Alternative way is using CString and transfer ownership.
+        // unsafe { self.0.as_mut() }.map(|table| {
+        //     let c_value = CString::new(value).unwrap();
+        //     // table.value.len = c_value.as_bytes().len();
+        //     table.value.len = c_value.len();
+        //     table.value.data = c_value.into_raw() as *mut u_char;
+        // });
+    }
+
+    pub fn set_key(&mut self, key: &str) {
+        // we can use pool to allocate and then copy
+        unsafe { self.0.as_mut() }.map(|table| {
+            table.key.len = key.len() as _;
+            table.key.data = crate::ffi::str_to_uchar(self.1, key);
+            table.lowcase_key = crate::ffi::str_to_uchar(self.1, key.to_lowercase().as_str());
+        });
+    }
+
+    pub fn set_hash(&mut self, hash: ngx_uint_t) {
```

Comment:
Overall comment, we'll need doc strings for the API. I don't get what this is doing, is it setting the hash algorithm? Is hash a closed set of values, if so, an enum could be more expressive.

Reply:
This is magic from the C guide:

from https://nginx.org/en/docs/dev/development_guide.html#list

> For example, to mark HTTP output headers (which are stored as ngx_table_elt_t objects) as missing, set the hash field in ngx_table_elt_t to zero. Items marked in this way are explicitly skipped when the headers are iterated over.

it is trick to "delete" header (by marking hash =0 for that element)

Reply:
basically set_key, set_hash, set_value mimic the NGINX's API.